src/inc/msquic_winuser.h

```diff
@@ -19,6 +19,14 @@
 #ifndef _MSQUIC_WINUSER_
 #define _MSQUIC_WINUSER_
 
+#ifndef WIN32_LEAN_AND_MEAN
+#define WIN32_LEAN_AND_MEAN
+#endif
```

Comment:
This file is used in the Windows build, so we have to be careful this doesn't break any internal builds. I'm hesitant to make this change. Similarly, for the one below, if it's necessary.

Reply:
We define this for all builds already in the build system.

Reply:
That's **our** build, not the consumer's build. By putting something in this header, you're affecting the consumer.

Reply:
https://docs.microsoft.com/en-us/windows/win32/winsock/creating-a-basic-winsock-application

Its required if you include windows.h and ws2def.h in the same file.

Reply:
Well the consumer literally cannot use this header without that define. The header breaks without it, as documented above. I'll remove the security one though.

Reply:
Ok. You're signing up to fix the Windows build if this breaks. :)

Reply:
Are you worried about our internal builds breaking or public builds?

Also, this is a lot of the reason why many public libraries completely abstract away winsock. It really does become a nightmare, because Windows.h and winsock2 do not play along.

Reply:
This header is ingested by the Windows OS repo. Then internal Windows code includes it, and is built as a part of the Windows build. So it's possible to check in something to this file that caused the Windows build to fail.